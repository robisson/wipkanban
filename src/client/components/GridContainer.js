// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import { withStyles } from "@material-ui/core/styles";

const styles: Object = {
  root: {
    flexGrow: 1
  },
  gridContainer: {
    marginTop: 68
  }
};

type Props = {
  classes: Object,
  children: React.Node
};

const GridContainer = ({ classes, children }: Props) => {
  return (
    <Grid container spacing={0} className={classes.gridContainer}>
      {children}
    </Grid>
  );
};

export default withStyles(styles)(GridContainer);
