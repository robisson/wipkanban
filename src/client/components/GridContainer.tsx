import * as React from "react";
import Grid from "@material-ui/core/Grid";
import { withStyles } from "@material-ui/core/styles";

const styles: any = () => {
  return {
    root: {
      flexGrow: 1
    },
    gridContainer: {
      marginTop: 68
    }
  };
};

interface IGridContainerProps {
  classes: any;
  children: any;
}

const GridContainer: React.StatelessComponent<any> = ({
  classes,
  children
}: IGridContainerProps) => {
  return (
    <Grid container spacing={0} className={classes.gridContainer}>
      {children}
    </Grid>
  );
};

export default withStyles(styles)(GridContainer);
