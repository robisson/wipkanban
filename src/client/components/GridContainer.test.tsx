import React from "react";
import GridContainer from "./GridContainer";
import { createShallow } from "@material-ui/core/test-utils";
import { ShallowWrapper } from "enzyme";

describe("TeamList Component", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should render its children", () => {
    let wrapper: ShallowWrapper = shallow(
      <GridContainer>
        <div>Hello</div>
      </GridContainer>
    );
    expect(wrapper.find("div").text()).toEqual("Hello");
    expect(wrapper).toMatchSnapshot();
  });
});
