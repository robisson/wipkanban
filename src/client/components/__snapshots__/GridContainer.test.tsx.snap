// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamList Component should render its children 1`] = `
<WithStyles(Grid)
  className="GridContainer-gridContainer-2"
  container={true}
  spacing={0}
>
  <div>
    Hello
  </div>
</WithStyles(Grid)>
`;
